src/runtime/index.ts

```diff
@@ -268,3 +265,15 @@ export function goto(href: string, opts = { replaceState: false }) {
 		window.location.href = href;
 	}
 }
+
+export function preloadRoutes(pathnames: string[]) {
+	if (!routes) throw new Error(`You must call init() first`);
+
+	return routes
+		.filter(route => {
+			return !pathnames || pathnames.some(pathname => route.pattern.test(pathname));
```

Comment:
Will this take care of `4xx` and `5xx`, or is that to be added at a later stage?

Reply:
Just me being forgetful 😬 thanks! fix incoming